.github/workflows/tests-on-release.yml

```diff
@@ -0,0 +1,25 @@
+name: 'Release checks run'
+run-name: ${{ github.event_name == 'schedule' && 'scheduled' || github.event_name == 'workflow_dispatch' && 'on demand'  || github.event_name == 'release' && 'release' || '' }} checks for ${{ inputs.tag != '' && inputs.tag || github.ref_name }}
```

Comment:
Worth adding `( ... )` for better clarity to visually isolate the possible value here.